cms/articles/tests/test_models.py

```diff
@@ -233,26 +234,203 @@ def test_cite_this_page_is_not_shown_when_unticked(self):
+                mock_hrf = mock_response[ind + 1].split("/", 1)
+        mock_breadcrumb = mock_hrf[1][:-1]
+        mock_breadcrumb = mock_breadcrumb.split("/")
+        mock_breadcrumb = list(filter("".__ne__, mock_breadcrumb))
+        return mock_breadcrumb
+
+    def test_pagination_is_not_shown(self):
+        response = self.client.get(self.basic_page_url)
+        expected = 'class="ons-pagination__link"'
+        self.assertNotContains(response, expected)
+
+
+class PreviousReleasesWithoutPaginationTestCase(TestCase):
+    # PREVIOUS_RELEASES_PER_PAGE is default value 10
+    total_batch = 9
+
+    @classmethod
+    def setUpTestData(cls, cls_total_batch: int = total_batch):
+        cls.article_series = ArticleSeriesPageFactory(title="Article Series")
+        cls.articles = StatisticalArticlePageFactory.create_batch(cls_total_batch, parent=cls.article_series)
+        cls.previous_releases_url = cls.article_series.url + cls.article_series.reverse_subpage("previous_releases")
+
+    def test_page_content(self):
+        response = self.client.get(self.previous_releases_url)
+        mock_response = response.content.decode("utf-8").split("\n")
+        allowed_to_print = False
+        release_count = 0
+        for m in mock_response:
+            if " </ul>" in m:
+                allowed_to_print = False
+            if "<ul>" in m:
+                allowed_to_print = True
+            if allowed_to_print and "<li><a href=" in m:
+                release_count += 1
+        self.assertEqual(release_count, self.total_batch)
+
+    def test_breadcrumb_is_shown(self):
+        # ToDo
+        # this should include breadcrumb content comparison with the url but there is a bug
+        # mock_url = response.request['PATH_INFO']
+        # mock_url = mock_url.split("/")
+        # mock_url = list(filter(('').__ne__, mock_url))
+        # mock_breadcrumb = self.get_breadcrumbs(response)
+        # self.assertContains(response, expected)
+        # self.assertEqual(len(mock_url) - len(mock_breadcrumb),1)
+        # self.assertEqual(mock_url[0] , mock_breadcrumb[0])
+        # self.assertEqual(mock_url[1] , mock_breadcrumb[1])
+        response = self.client.get(self.previous_releases_url)
+        expected = 'class="ons-breadcrumbs__link"'
+        self.assertContains(response, expected)
+
+    def test_pagination_is_not_shown(self):
+        response = self.client.get(self.previous_releases_url)
+        expected = 'class="ons-pagination__link"'
+        self.assertNotContains(response, expected)
+
+
+@override_settings(PREVIOUS_RELEASES_PER_PAGE=3)
+class PreviousReleasesWithPaginationPage3TestCase(TestCase):
+    total_batch = 13
+    current_page_number = 3
+    PREVIOUS_RELEASES_PER_PAGE = 3
+    total_no_of_pages = math.ceil(total_batch / PREVIOUS_RELEASES_PER_PAGE)
+
+    @classmethod
+    def setUpTestData(cls, cls_total_batch: int = total_batch, cls_current_page_number: int = current_page_number):
```

Comment:
this is the method signature

```suggestion
    def setUpTestData(cls):
```

Django will not pass other args to it. With the signature updated, `cls_total_batch` -> `cls.total_batch`, `cls_current_page_number` -> `cls.current_page_number`

Reply:
Amended the values to hard coded so that only argument is cls :-(